phac-nml/irida-next
# [ENHC0010033] List samples within clone dialog

## What does this PR do and why?
- Added a scrollable list within the sample clone dialog to display the selected samples
- Added a spinner and loading text after cloning has been initiated
- Added a short description to the top of the sample transfer dialog (a continuation of ENHC0010032)

Fixes https://github.com/phac-nml/irida-next/issues/469.

## Screenshots or screen recordings
![image](https://github.com/phac-nml/irida-next/assets/325703/3a2aa158-37b2-4b9d-9cc4-d86b60763828)
![image](https://github.com/phac-nml/irida-next/assets/325703/fb253e2c-f497-40d2-9444-e69ea93fc075)

![image](https://github.com/phac-nml/irida-next/assets/325703/e81cbb9d-ca6f-4250-ba80-996bac7f4178)
![image](https://github.com/phac-nml/irida-next/assets/325703/c09b4949-0e33-4412-9c4b-ee57b52d83e4)

![image](https://github.com/phac-nml/irida-next/assets/325703/0a757b6d-d0bd-4fad-bb73-6eddb2dfdc51)
![image](https://github.com/phac-nml/irida-next/assets/325703/00f8e22c-dcd8-4351-b885-d86de8eece0c)

## How to set up and validate locally
1. Open a rails console and run the following:
 
```
project = Project.first
 
11.upto(10000) do |i|
  sample = Samples::CreateService.new(
    project.creator, project,
    { name: "#{project.namespace.parent.name}/#{project.name} Sample #{i}",
      description: "This is a description for sample #{project.namespace.parent.name}/#{project.name} Sample #{i}." 
    }
  ).execute
end
```
2. Login as admin and navigate to [http://localhost:3000/bacillus/bacillus-anthracis/outbreak-2022/-/samples](http://localhost:3000/bacillus/bacillus-anthracis/outbreak-2022/-/samples).
3. Select all the samples within the table.
4. Click the `Clone samples` button.
5. Verify the selected samples are displayed.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
